app/controllers/books_controller.rb

```diff
@@ -61,21 +61,27 @@ def update
 
   def list
     @book = Book.find_by(id: params[:book][:id])
-    @book.selected = params[:book][:selected]
-    if @book.save!
-      render json: {status: :success }
-    else
-      render json: {status: :failure }
+
+    if session[:selected_books]
```

Comment:
What if session[:selected_books] doesn't exist yet? Like this is the first book to be selected?